modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainer.java

```diff
@@ -1,83 +1,163 @@
 
+    private static final Integer INFLUXDB_PORT = 8086;
     private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("influxdb");
+    private static final int NO_CONTENT_STATUS_CODE = 204;
 
-    private static final String DEFAULT_TAG = "1.4.3";
 
-    @Deprecated
-    public static final String VERSION = DEFAULT_TAG;
+    @Getter
+    private String username = "any";
+    @Getter
+    private String password = "any";
 
+    /**
+     * Properties InfluxDB &lt;= 2
+     */
     private boolean authEnabled = true;
-
     private String admin = "admin";
-
     private String adminPassword = "password";
-
     private String database;
 
-    private String username = "any";
-
-    private String password = "any";
-
     /**
-     * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
+     * InfluxDB &gt;= 2
      */
-    @Deprecated
-    public InfluxDBContainer() {
-        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
-    }
+    @Getter
+    private String bucket = "test-bucket";
+    @Getter
+    private String organization = "test-org";
+    @Getter
+    private Optional<String> retention = Optional.empty();
+    @Getter
+    private Optional<String> adminToken = Optional.empty();
 
-    /**
-     * @deprecated use {@link InfluxDBContainer(DockerImageName)} instead
-     */
-    @Deprecated
-    public InfluxDBContainer(final String version) {
-        this(DEFAULT_IMAGE_NAME.withTag(version));
+    private final boolean isAtLeastMajorVersion2;
+
+
+    public InfluxDBContainer(final String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
     }
 
     public InfluxDBContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
 
-        waitStrategy =
-            new WaitAllStrategy()
-                .withStrategy(Wait.forHttp("/ping").withBasicCredentials(username, password).forStatusCode(204))
-                .withStrategy(Wait.forListeningPort());
+        logger().info("Starting an InfluxDB container using [{}]", dockerImageName);
+        this.waitStrategy = new WaitAllStrategy()
+            .withStrategy(
+                Wait
+                    .forHttp("/ping")
+                    .withBasicCredentials(this.username, this.password)
+                    .forStatusCode(NO_CONTENT_STATUS_CODE)
+            )
+            .withStrategy(Wait.forListeningPort());
 
-        addExposedPort(INFLUXDB_PORT);
+        this.isAtLeastMajorVersion2 =
+            new ComparableVersion(dockerImageName.getVersionPart()).isGreaterThanOrEqualTo("2.0.0");
+        this.addExposedPort(INFLUXDB_PORT);
     }
 
+    /**
+     * Sets the InfluxDB environment variables based on the version
+     */
     @Override
     protected void configure() {
-        addEnv("INFLUXDB_ADMIN_USER", admin);
-        addEnv("INFLUXDB_ADMIN_PASSWORD", adminPassword);
+        if (isAtLeastMajorVersion2) {
```

Comment:
latest should be included as a version 2?

Reply:
According to the [Dockerhub description](https://hub.docker.com/_/influxdb?tab=description), yes:
```
The latest tag for this image now points to the latest released implementation of InfluxDB 2.x. If you are using the latest tag and would like to stay on the InfluxDB 1.x line, please update your environment to reference the 1.8 tag.
```